tfconfig/provider_ref.go

```diff
@@ -1,9 +1,85 @@
 package tfconfig
 
+import (
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
 // ProviderRef is a reference to a provider configuration within a module.
 // It represents the contents of a "provider" argument in a resource, or
 // a value in the "providers" map for a module call.
 type ProviderRef struct {
 	Name  string `json:"name"`
 	Alias string `json:"alias,omitempty"` // Empty if the default provider configuration is referenced
 }
+
+type ProviderRequirement struct {
+	Source             string   `json:"source"`
+	VersionConstraints []string `json:"version_constraints"`
```

Comment:
It looks like on `ModuleCall` we called this `Version` and made it just a string, though I guess this one needs to stay an array because multiple different bits of the config contribute to it :confused: .

Making this consistent would make it harder to use, so... I guess we will live with it. :grinning: I'd suggest putting `omitempty` on it to make the JSON output more compact where there are no constraints to list, though.

Reply:
The difference is that each ModuleCall should only have a single version constraint per provider, but the whole module tree can result in a list of constraints. 

`omitempty` is _really obvious_ in hindsight, thank you for the prompt.